bsip-0047.md

```diff
@@ -0,0 +1,230 @@
+* `optional<account_id_type> worker_voting_account`
+
+**Validation:**
+
+* These extensions must not be present before the protocol change activation date, neither in transactions nor in proposals.
+* If any extension is present, it must contain the ID of an existing account.
+* If any extension is present, `account_options.voting_account == GRAPHENE_PROXY_PER_CATEGORY_ACCOUNT`
+
+
+### `account_object`(Existing)
+
+**Note:** this is an implementation detail and meant to be a hint for later development. The actual implementation may differ.
+
+The account object shall receive these new fields:
+
+* `flat_set<vote_id_type> committee_votes`
+* `flat_set<vote_id_type> witness_votes`
+* `flat_set<vote_id_type> worker_votes`
+
+`account_create_operation` and `account_update_operation` shall be modified to create these sets by filtering the `account_options.votes` field appropriately. `account_update_votes_operation` shall also create these sets by filtering the `votes_to_add` and `votes_to_remove` fields appropriately.
+
+
+### `account_update_votes_operation` (New)
+
+The following is an example implementation.
+
+```
+struct account_update_votes_operation : public base_operation
+{
+	asset fee;
+
+	/// The account to update
+	account_id_type account;
+
+	/// New account options
+	flat_set<vote_id_type>    votes_to_add;
+	flat_set<vote_id_type>    votes_to_remove;
+
+	// A new voting account to set
+	optional<account_id_type> committee_voting_account
+	optional<account_id_type> witness_voting_account
+	optional<account_id_type> worker_voting_account
+
+	// A new number of witness_votes to set
+	optional<uint16_t> num_witness;
+
+	// A new number of committee_votes to set
+	optional<uint16_t> num_committee;
+
+	// For future extensions (see account_update_operation)
+	extensions_type extensions;
+};
+```
+
+Changes to direct votes can be specified by means of a *list* of vote identifiers to be added or removed. This has multiple advantages:
+
+* Reduces the size of the operation compared to replacing the entire slate
+* Allows to easily compare votes before and after the operation
+
+**Validation checks:**
+
+Many of the validation checks are dependent on the merger of the _proposed_ values in the operation with the account's _current_ voting values on the blockchain.
+
+_General validation checks_
+* `account` must be a valid account ID, must exist, must authorize the operation, must pay fee
+* `committee_voting_account`, if present, must be a valid account ID, must exist
+* `witness_voting_account`, if present, must be a valid account ID, must exist
+* `worker_voting_account`, if present, must be a valid account ID, must exist
+* Validate the existence of each vote to add and remove
+
+_Committee validation checks_
+* Let the new _committee_ slate consist of the current _commitee_ slate plus the _committee_ `votes_to_add` minus the _committee_ `votes_to_remove`
+* If `account_options.committee_voting_account == GRAPHENE_PROXY_TO_SELF_ACCOUNT`
```

Comment:
I think what you mean is
* If committee_voting_account is present, let new_committee_account = committee_voting_account
* otherwise let new_committee_account = account_options.committee_voting_account

Reply:
Indeed, thank you.  I'd like to say that I was thinking about some corner case but it just looks wrong now.  The text has been updated.